service/storage_service.cc

```diff
@@ -6020,6 +6020,24 @@ void storage_service::init_messaging_service(bool raft_topology_change_enabled)
                 };
             });
         });
+        ser::storage_service_rpc_verbs::register_raft_pull_general_snapshot(&_messaging.local(), [handle_raft_rpc] (raft::server_id dst_id, raft_general_snapshot_pull_params params) {
```

Comment:
It's safer not to register a handler for this RPC if auth-v2 experimental feature is not enabled.

Reply:
Don't see how it's safer, can something fail during registration?. This rpc will be also used by service levels and potentially other features so there is no isolated condition it can be enabled upon.

Reply:
Hmm, if it's going to replace the topology snapshot RPC and be generic then it indeed makes less sense.

In general, we should avoid any external side effects related to some experimental feature if it is not enabled. It's better not to have to think what could potentially happen if some other node sends this RPC when this node has this feature marked as disabled. Moreover, if we find that we accidentally started to use it then we might run into compatibility problem.

Better to protect the experimental verb from two sides (i.e. on the sender and the receiver side).